types/queries.go

```diff
@@ -23,16 +23,23 @@ func RustQuery(querier Querier, binRequest []byte, gasLimit uint64) QuerierResul
 	var request QueryRequest
 	err := json.Unmarshal(binRequest, &request)
 	if err != nil {
-		return ToQuerierResult(nil, UnsupportedRequest{err.Error()})
+		return QuerierResult{
+			Err: &SystemError{
+				InvalidRequest: &InvalidRequest{
+					Err:     err.Error(),
+					Request: binRequest,
+				},
+			},
+		}
```

Comment:
If I understand this diff 👆 correctly, this was a bug before because an error parsing bytes to `QueryRequest` was treated as an `UnsupportedRequest`. But this can be any kind of parsing error.

Reply:
Yes, it does seem a more proper error code this way